diff --git a/core/dnsserver/zdirectives.go b/core/dnsserver/zdirectives.go
index f6496280e..693f01d31 100644
--- a/core/dnsserver/zdirectives.go
+++ b/core/dnsserver/zdirectives.go
@@ -63,0 +64 @@ var Directives = []string{
+	"edgecdnx",
diff --git a/core/plugin/zplugin.go b/core/plugin/zplugin.go
index 476d8c1dc..5f2fc9328 100644
--- a/core/plugin/zplugin.go
+++ b/core/plugin/zplugin.go
@@ -63,4 +63,5 @@ import (
 	_ "github.com/coredns/coredns/plugin/tsig"
 	_ "github.com/coredns/coredns/plugin/view"
 	_ "github.com/coredns/coredns/plugin/whoami"
+	_ "github.com/EdgeCDN-X/edgecdnx-plugin"
 )
diff --git a/go.mod b/go.mod
index 1506e553e..d3dcb7dd8 100644
--- a/go.mod
+++ b/go.mod
@@ -5,6 +5,7 @@ module github.com/coredns/coredns
 go 1.25.0
 
 require (
+	github.com/EdgeCDN-X/edgecdnx-plugin v0.0.0
 	github.com/Azure/azure-sdk-for-go v68.0.0+incompatible
 	github.com/Azure/go-autorest/autorest v0.11.30
 	github.com/Azure/go-autorest/autorest/azure/auth v0.5.13
@@ -204,3 +205,5 @@ require (
 	sigs.k8s.io/structured-merge-diff/v6 v6.3.0 // indirect
 	sigs.k8s.io/yaml v1.6.0 // indirect
 )
+
+replace github.com/EdgeCDN-X/edgecdnx-plugin => ../
\ No newline at end of file
diff --git a/plugin.cfg b/plugin.cfg
index 0d735cef2..7ec0c61f9 100644
--- a/plugin.cfg
+++ b/plugin.cfg
@@ -24,6 +24,7 @@ metadata:metadata
 geoip:geoip
 cancel:cancel
 tls:tls
+proxyproto:proxyproto
 quic:quic
 grpc_server:grpc_server
 https:https
@@ -71,0 +73 @@ loop:loop
+edgecdnx:github.com/EdgeCDN-X/edgecdnx-plugin
